apps/extension/src/routes/page/onboarding/generate.tsx

```diff
@@ -14,20 +14,74 @@ import { generateSelector } from '../../../state/seed-phrase/generate';
 import { usePageNav } from '../../../utils/navigate';
 import { PagePath } from '../paths';
 import { WordLengthToogles } from '../../../shared/containers/word-length-toogles';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { fetchBlockHeight } from '../../../hooks/full-sync-height';
+import { localExtStorage } from '../../../storage/local';
+import { sample } from 'lodash';
+
+const fetchBlockHeightWithFallback = async (endpoints: string[]): Promise<number> => {
+  if (endpoints.length === 0) {
+    throw new Error('All RPC endpoints failed to fetch the block height.');
+  }
+
+  // Randomly select an rpc endpoint from the chain registry
+  const randomEndpoint = sample(endpoints);
+
+  try {
+    const walletCreationBlockHeight = await fetchBlockHeight(randomEndpoint!);
```

Comment:
issue: In the other repo, we now have an eslint rule for us to avoid using `!`. Can you have an error guard above instead?

```ts
  if (!randomEndpoint) {
    throw new Error('No RPC endpoints found.');
  }
```

Reply:
added type gaurd

Reply:
references to the eslint rules implemented in https://github.com/penumbra-zone/web/pull/1721 that still need to be propagated to this repository.